slsim/lens_pop.py

```diff
@@ -1,378 +1,70 @@
-        :type sn_type: str
-        :param sn_absolute_mag_band: Band used to normalize to absolute magnitude
-        :type sn_absolute_mag_band: str or `~sncosmo.Bandpass`
-        :param sn_absolute_zpsys: Optional, AB or Vega (AB default)
-        :type sn_absolute_zpsys: str
-        :param los_config: configuration for line of sight distribution
-        :type los_config: LOSConfig instance
-        :param sn_modeldir: sn_modeldir is the path to the directory containing files
-         needed to initialize the sncosmo.model class. For example,
-         sn_modeldir = 'C:/Users/username/Documents/SALT3.NIR_WAVEEXT'. These data can
-         be downloaded from https://github.com/LSST-strong-lensing/data_public .
-         For more detail, please look at the documentation of RandomizedSupernovae
-         class.
-        :type sn_modeldir: str
+        Args:
+            deflector_population (DeflectorsBase): Deflector population as an instance of a DeflectorsBase subclass.
+            source_population (SourcePopBase): Source population as an instance of a SourcePopBase subclass
+            cosmo (Optional[Cosmology], optional): AstroPy Cosmology instance. If None, defaults to flat LCDM with h0=0.7 and Om0=0.3.
+                                                   Defaults to None.
+            lightcurve_time (Optional[np.ndarray], optional): Lightcurve observation time array in units of days. Defaults to None.
+            sn_type (Optional[str], optional): Supernova type (Ia, Ib, Ic, IIP, etc.). Defaults to None.
+            sn_absolute_mag_band (Optional[Union[str,sncosmo.Bandpass]], optional): Band used to normalize to absolute magnitude.
+                                                                                    Defaults to None.
+            sn_absolute_zpsys (Optional[str], optional): Zero point system, either AB or Vega, with None defaulting to AB.
+                                                         Defaults to None.
+            los_config (Optional[LOSConfig], optional): Configuration for line of sight distribution. Defaults to None.
+            sn_modeldir (Optional[str], optional): sn_modeldir is the path to the directory containing files needed to initialize
+                                                   the sncosmo.model class. For example, sn_modeldir =
+                                                   'C:/Users/username/Documents/SALT3.NIR_WAVEEXT'. These data can be downloaded
+                                                   from https://github.com/LSST-strong-lensing/data_public. For more detail,
+                                                   please look at the documentation of RandomizedSupernovae class. Defaults to None.
         """
+
         super().__init__(
-            sky_area,
-            cosmo,
-            lightcurve_time,
-            sn_type,
-            sn_absolute_mag_band,
-            sn_absolute_zpsys,
-            sn_modeldir,
+            sky_area=None,
```

Comment:
there should be a sky_area quantity in LensPop. This should be particularly different then the sky area of the deflector and/or source population to make it easier to create larger sky area realizations

Reply:
How is this the case in the current LensPop implementation? As far as I understand, if we provide a deflector and/or source population sky_area, the LensPop sky_area is ignored. Have I misunderstood this?

Reply:
The LensPop sky_area is the ultimate area over which strong lensing populations are simulated. If the source and/or deflector sky areas are different, then the multiplicative factor between the final sky area and the sub-population is computed and then the individual deflector/source populations are up (or down) sampled from.
I.e. we do not need 100 billion of galaxies (half the sky or so) to simulate lenses on half the sky. As long as we have a much smaller but still representative area, we can just re-use the much smaller populations multiple times (randomly)

Reply:
Thanks, helped me realise I'd missed the f_sky in the lens / deflector factors. Makes sense, will fix asap.